### PR TITLE
HL-343 | Browser tests

### PR DESCRIPTION
**Pull request by Nizar-Rahme**
## Description :sparkles:
- Implement e2e tests for the applicant side

### CODE REVIEW

File: frontend/benefit/applicant/browser-tests/roles/company.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Role } from 'testcafe';
+import login from '../page-modal/login';
+import { getFrontendUrl } from '../utils/url.utils';
+
+const url = getFrontendUrl('/');
+
+const company = Role(url, async (t) => {
```
Comment (nenonja) [inline frontend/benefit/applicant/browser-tests/roles/company.ts:7]: Role seems to be a clean way to do the logging! I would suggest you to use also MOCK_FLAG (env variable) to completely mock the login from backend, it makes testing and developing easier. Not sure does your backend support it however...
Perhaps I could start using this roles as well.

File: frontend/package.json [hunk 2]
```diff
@@ -33,6 +33,8 @@
     "test:coverage": "npx lerna run test:coverage --stream --concurrency 1",
     "bf-applicant:test": "yarn --cwd benefit/applicant test",
     "bf-handler:test": "yarn --cwd benefit/bf-handler test",
+    "bf-applicant:browser-test": "yarn --cwd benefit/applicant browser-test",
+    "bf-applicant:browser-test:ci": "yarn --cwd benefit/applicant browser-test:ci",
```
Comment (nenonja) [inline frontend/package.json:37]: you probably have to add some certificate flags to make the tests work on review environment. See the same script on kesaseteli-employer

File: .github/workflows/bf-review.yml [hunk 3]
```diff
@@ -152,7 +152,26 @@ jobs:
...
+      fail-fast: false
+      matrix:
+        service: ["bf-appl"]
+        include:
+          - service: bf-appl
+            dir: benefit/applicant
+    steps:
+      - name: Run Acceptance Tests for ${{ matrix.service }}
+        id: testcafe
+        run: yarn --cwd ${{ matrix.dir }} browser-test:ci
```
Comment (nenonja) [inline .github/workflows/bf-review.yml:172]: the script seems to be missing from benefit/applicat package.json

**Review by nenonja — COMMENTED** [review]
Great start and this class approach gave me some good ideas too how to implement component apis! I assume that browser tests are nwo working locally, but making them to work in review environment needs some work too..

**Comment by Nizar-Rahme** [comment]
For some reason the pipeline is always failing here, I tried to open a new PR with the same changes and the error was gone 🤷 .

File: frontend/benefit/applicant/browser-tests/roles/company.ts (hunk 1 above)
Comment (Nizar-Rahme) [reply frontend/benefit/applicant/browser-tests/roles/company.ts:7]: I added the MOCK flag and disabled the authentication, it worked well in the local env, but I am getting CORS errors I couldn't manage to work around.

File: frontend/package.json (hunk 2 above)
Comment (Nizar-Rahme) [reply frontend/package.json:37]: Copy-pasted from Kesäseteli 🙈

**Comment by Nizar-Rahme** [comment]
Addressed some of the comments in https://github.com/City-of-Helsinki/yjdh/pull/580
Closing thin PR in favor of https://github.com/City-of-Helsinki/yjdh/pull/580

**Pull request closed without merging**